packages/server/src/api/rest/index.ts

```diff
@@ -1178,18 +1182,31 @@ class RequestHandler extends APIHandlerBase {
         return r.toString();
     }
 
-    private makeIdFilter(idField: string, idFieldType: string, resourceId: string) {
-        return { [idField]: this.coerce(idFieldType, resourceId) };
+    private makeIdFilter(idFields: IdField[], resourceId: string) {
+        if (idFields.length === 1) {
+            return { [idFields[0].name]: this.coerce(idFields[0].type, resourceId) };
+        } else {
+            return {
+                [idFields.map((idf) => idf.name).join('_')]: idFields.reduce(
```

Comment:
Prisma actually allows to you name a compound id field like @@id("myCompountId", ...). When used that way, the default compound key formatting actually won't work.

Supporting it will require carrying more information about @@id in model metadata and will require some changes in the code gen part. I can make a separate follow up PR for it later.

Reply:
Good catch. Should we keep it like this for now in this PR then?

Reply:
Yes, I think we can keep it like this for now.